pkg/frontend/frontend.go

```diff
@@ -503,3 +504,37 @@ func reply(log *logrus.Entry, w http.ResponseWriter, header http.Header, b []byt
 		_, _ = w.Write([]byte{'\n'})
 	}
 }
+
+func frontendOperationResultLog(log *logrus.Entry, method string, err error) {
+	log = log.WithFields(logrus.Fields{
+		"LOGKIND":       "frontendqos",
+		"resultType":    utillog.SuccessResultType,
+		"operationType": method,
+	})
+
+	if err == nil {
+		log.Info("front end operation succeeded")
+		return
+	}
+
+	if err != nil {
+		switch err := err.(type) {
+		case *api.CloudError:
+			log = log.WithField("resultType", utillog.UserErrorResultType)
+		case statusCodeError:
+			if int(err) < 300 {
+				log.Info("front end operation succeeded")
+				return
+			} else if int(err) < 500 {
+				log = log.WithField("resultType", utillog.UserErrorResultType)
+			} else {
+				log = log.WithField("resultType", utillog.ServerErrorResultType)
+			}
+		default:
+			log = log.WithField("resultType", utillog.ServerErrorResultType)
+		}
+	}
+
+	log = log.WithField("errorDetails", err.Error())
```

Comment:
I'm not too familiar with this log api, is this appending the error in addition to the `operationType` etc set on line 509?

Reply:
Yes, so the errorDetails will be blank until it gets to this point and will append it to the log. Similarly done [here ](https://github.com/Azure/ARO-RP/blob/2ca94a547ffa2522da45967abbd400b35a6c3ce8/pkg/backend/openshiftcluster.go#L340) for the back end.

Reply:
very cool, thanks for the additional example!